lnwallet/channel.go

```diff
@@ -1704,6 +1705,134 @@ func (lc *LightningChannel) SignNextCommitment() ([]byte, error) {
 	return sig, nil
 }
 
+// ReceiveReestablish is used to handle the remote channel reestablish message
+// and generate the set of updates which are have to be sent to remote side
+// to synchronize the states of the channels.
+func (lc *LightningChannel) ReceiveReestablish(msg *lnwire.ChannelReestablish) (
+	[]lnwire.Message, error) {
+
+	lc.Lock()
+	defer lc.Unlock()
+
+	// As far we store on last commitment transaction we should rely on the
+	// height of the commitment transaction in order to calculate the length.
+	numberRemoteCommitments := lc.remoteCommitChain.tip().height + 1
+
+	// Number of the revocations might be calculated as the height of the
+	// commitment transactions which will be revoked next minus one. And plus
+	// one because height starts from zero.
+	numberRemoteRevocations := lc.remoteCommitChain.tail().height - 1 + 1
+
+	var updates []lnwire.Message
+	commitmentChainDiff := msg.NextLocalCommitmentNumber - numberRemoteCommitments
+	if commitmentChainDiff == 0 {
+		// If remote side expects as receive commitment which we already
+		// consider as last, than it means that they aren't received our
+		// last commit sig message.
+		commitment := lc.remoteCommitChain.tip()
+		chanID := lnwire.NewChanIDFromOutPoint(lc.channelState.ChanID)
+		for _, htlc := range commitment.outgoingHTLCs {
```

Comment:
Iterating over the `outgoingHTLCs` isn't sufficient. This only stores the HTLC add messages sent by either side. In the case of a retransmission, we need to send the _full_ delta over to the remote party. This includes any settle, add, and fail messages sent. 

In order to accomplish this, we need to _write to disk_ each time within the `SignNextCommitment` method. The contents written will include the _exact_ set of _new_ messages sent between this commitment and the last which were ACK'd and included in this new commitment state. Additionally, we'll need to also persist the _counters_ within the update log to disk as well.